src/data/finset/basic.lean

```diff
@@ -11,19 +11,92 @@ import tactic.nth_rewrite
 /-!
 # Finite sets
 
-mathlib has several different models for finite sets,
-and it can be confusing when you're first getting used to them!
+What are `finset`s in mathlib?
+
+Terms of type `finset α` are subsets of `α` which are finite.
+In Lean, `finset α` is implemented to be 2 pieces of data:
+
+  1) `val` is a `multiset α` of elements
+  2) `no_dup` is a proof that `val` has no duplicates.
+
+Finsets in Lean are constructive since you must exhibit
+a list of their underlying elements to use them.
+In particular, you must show that any function that
+uses the data of the underlying list does not depend on its ordering.
+In practice, however, this is handled on the multiset level by
+multiset API, so you will never have to do this explicitly.
+
+Finsets give a basic foundation for defining finite sums and products over types, in the form of
+  `∑ i in (s : finset α), f i` and `∏ i in (s : finset α), f i`.
+Lean refers to these operations as `big_operator`s.
+More information can be found in `algebra/big_operators/basic`.
+Confusingly, there are three different ways to sum over finite things: {finset, fintype, finsupp}.
```

Comment:
There's a fourth, dfinsupp

Reply:
Oh no. I didn't even know about that one.